### ISSUE TITLE
Flash / Flash_CLI Examples Have Misleading CLKDIV 

### ISSUE THREAD

**Issue by Jake-Carter** [Analog-Devices-MSDK/msdk#287]
TLDR:  All Flash and Flash_CLI examples for all microcontrollers should checked and updated to match the AI85 Flash example to remove this misleading clock divisor.

The Flash example had a misleading clock divisor in its flash initialization routine.  I fixed this for the AI85 and reworked the Flash example.  See [flash_init()](https://github.com/Analog-Devices-MSDK/msdk/blob/1da6d2de37c0d88387ba0820fe73bbd74be123e3/Examples/MAX78000/Flash/main.c#L115)

**Recommended Flash initialization:**
```C
void flash_init(void)
{
    /*
    The NVIC_SetRAM function below sets ISRs to execute out of RAM.
    All functions modifying flash contents are set to execute out of RAM
    with the (section(".flashprog")) attribute.  Therefore,
    
    If:
    - An FLC function is in the middle of execution (from RAM)
    ... and...
    - An interrupt triggers an ISR which executes from Flash
    ... Then a hard fault will be triggered.  
    
    FLC functions should be:
    1) Executed from a critical code block (interrupts disabled)
    or
    2) ISRs should be set to execute out of RAM with NVIC_SetRAM()
    It's preferable to use #1 but the Flash Controller can also 
    generate interrupts of its own (as shown in this example), 
    in which case use #2 for the FLC ISRs.
    */
    NVIC_SetRAM(); // Execute ISRs out of SRAM
    MXC_NVIC_SetVector(FLC0_IRQn, FLC0_IRQHandler); // Assign ISR
    NVIC_EnableIRQ(FLC0_IRQn); // Enable interrupt
    __enable_irq();

    // Clear and enable flash programming interrupts
    MXC_FLC_EnableInt(MXC_F_FLC_INTR_DONEIE | MXC_F_FLC_INTR_AFIE);
    isr_flags = 0;
    isr_cnt = 0;
}
```

However, many other micros still have the old misleading clock divider **including the Flash_CLI examples**.  (Ex:  https://github.com/Analog-Devices-MSDK/msdk/blob/main/Examples/MAX32670/Flash/main.c#L192)

```C
void flash_init(void)
{
    // Set flash clock divider to generate a 1MHz clock from the APB clock
    // APB clock is 54MHz on the real silicon
    MXC_FLC0->clkdiv = 24;

    MXC_FLC_ClearFlags(0x3);
    // Setup and enable interrupt
    MXC_NVIC_SetVector(FLC0_IRQn, FLC0_IRQHandler);
    NVIC_EnableIRQ(FLC0_IRQn);
    __enable_irq();
}
```

The clock divider is **incorrect _and_ unnecessary**.  Our FLC drivers set the clock divider to match the 1MHz requirement of the flash controller in [`MXC_prepare_flc`](https://github.com/Analog-Devices-MSDK/msdk/blob/1da6d2de37c0d88387ba0820fe73bbd74be123e3/Libraries/PeriphDrivers/Source/FLC/flc_reva.c#L91), which is called before initiating any flash operations.

```C
static int MXC_prepare_flc(mxc_flc_reva_regs_t *flc)
{
    /* Check if the flash controller is busy */
    if (MXC_busy_flc(flc)) {
        return E_BUSY;
    }

    // Set flash clock divider to generate a 1MHz clock from the APB clock
    flc->clkdiv = SystemCoreClock / 1000000;

    /* Clear stale errors */
    if (flc->intr & MXC_F_FLC_REVA_INTR_AF) {
        flc->intr &= ~MXC_F_FLC_REVA_INTR_AF;
    }

    /* Unlock flash */
    flc->ctrl = (flc->ctrl & ~MXC_F_FLC_REVA_CTRL_UNLOCK) | MXC_S_FLC_REVA_CTRL_UNLOCK_UNLOCKED;

    return E_NO_ERROR;
}
```
